contracts/AllowanceTarget.sol

```diff
@@ -16,8 +16,11 @@ contract AllowanceTarget is IAllowanceTarget, Pausable, Ownable {
 
     constructor(address _owner, address[] memory trustedCaller) Ownable(_owner) {
         uint256 callerCount = trustedCaller.length;
-        for (uint256 i = 0; i < callerCount; ++i) {
+        for (uint256 i = 0; i < callerCount; ) {
             authorized[trustedCaller[i]] = true;
+            unchecked {
+                ++i;
+            }
```

Comment:
I feel that this is a bit too much for gas optimization. I would prefer trading this gas cost for coding convention and readability.

Reply:
I feel like this pattern is more difficult to follow too. Unless documented properly, devs could misuse or break the pattern when copy-pasting it in the future.

Reply:
Ok, let's skip this refactoring for now. It's kind of common for me but I do think it's more error-prone.

Reply:
Fixed.